### ISSUE TITLE
Collection testing bug

### ISSUE THREAD

**Issue by joachimwolff** [galaxyproject/planemo#630]
If a collection is defined for the outputs of a tool:

```
<outputs>
    <collection name="intermediate_results" label="Intermediate results" type="list">
        <data name="some_name_tag" format="fastq" label="Fancy_output" from_work_dir="some_name_tag.fastq" />
        <data name="some_name_tag_2" format="fastq" label="Fancy_output_2" from_work_dir="some_name_tag_2.fastq" />
    </collection>
</outputs>
```

and the testing is written wrong in the following way:

```
<tests>
    <test>
        <output name="some_name_tag" file="some_file" lines_diff="2"/>
        <output name="some_name_tag_2" file="some_file_2" lines_diff="2"/>
    </test>
</tests>
```

instead of:

```
<tests>
    <test>
         <output_collection name="intermediate_results">
            <element name="some_name_tag" file="some_file" lines_diff="2"/>
            <element name="some_name_tag_2" file="some_file" lines_diff="2"/>
        </output_collection>
    </test>
</tests>
```

than it happens that `planemo lint` says every thing is correct which is obviously wrong. The `planemo test` command executes it and fails. But the reason for the fail is now weird: Instead of crashing because some file could not be found, it fails because the name of the output files and the content of files are somehow random assigned to each other. And than the testcases fail. In the above example it would mean that the correct computed 'some_name_tag'-file would maybe have the content of 'some_name_tag_2'-file. 

It would be nice if the linter could see that the test case is defined wrong, or that the testing looks for the collection and if it is not there it should raise an error instead of reassigning file names and content randomly without any warning. 

I hope I explained my issue in a way you can reproduce. If not I have a concrete implementation of this in my account and I could explain what you have to do if requested.

**Comment by jmchilton** [comment]
Yeah - that is definitely a current shortcoming of the linter. This is also related to https://github.com/galaxyproject/planemo/issues/627. It would be nice to tackle both of these together. I'll try to find some time to do that.

**Comment by jmchilton** [comment]
I believe this is fixed in the master branch and the fix will be included whenever planemo 0.39.0 is released. Thanks for the detailed bug report!
